report_page_orientation/models/ir_actions_report.py

```diff
@@ -0,0 +1,235 @@
+            match_klass = (
+                "//div[contains(concat(' ', normalize-space(@class), ' '), ' {} ')]"
+            )
+            section_elements = root.xpath(
+                match_klass.format("rotated-page-orientation")
+            )
+            if not section_elements:
+                new_bodies.append(body)
+                continue
+
+            # if yes, pre-render these sections separately, to see how many pages they are
+            _logger.info("Body %s: %s sections", i, len(section_elements))
+            bodies_with_rotation += 1
+            sections = []
+            for section_element in section_elements:
+                content = lxml.html.tostring(section_element, encoding="unicode")
+                section_pdf_content = super(IrActionsReport, self)._run_wkhtmltopdf(
+                    [content],
+                    header=header,
+                    footer=footer,
+                    landscape=not landscape,
+                    specific_paperformat_args=specific_paperformat_args,
+                    set_viewport_size=set_viewport_size,
+                )
+                section_pdf_stream = io.BytesIO(section_pdf_content)
+                section_pdf = PdfFileReader(section_pdf_stream)
+                section_pdf_pages = section_pdf.getNumPages()
+                sections.append(
+                    {
+                        "num_pages": section_pdf_pages,
+                        "element": section_element,
+                        "content": content,
+                    }
+                )
+
+            # within the body, replace such a section with a number of page breaks equal
+            #     to each section that is in between. mark such a page break with a class. also
+            #     mark the first one per section.
+            for section in sections:
+                placeholderpage = lxml.html.fromstring(
+                    "<h1 style='page-break-before: always;'>PLACEHOLDER</h1>"
+                )
+                morepage = lxml.html.fromstring(
+                    "<div style='page-break-before: always;'>MORE</div>"
+                )
+                page_break = lxml.html.fromstring(
+                    "<div style='page-break-before: always;'/>"
+                )
+                element = section["element"]
+                parent = element.getparent()
+                parent.replace(element, placeholderpage)
+                placeholderpage.addnext(page_break)
+                num_pages = section["num_pages"]
+                if num_pages > 1:
+                    for _i in range(num_pages - 1):
+                        page_break.addprevious(deepcopy(morepage))
+            if sections:
+                body = lxml.html.tostring(root, encoding="unicode")
+            new_bodies.append(body)
+
+        # If no change, just return super.
+        if not bodies_with_rotation:
+            return super(IrActionsReport, self)._run_wkhtmltopdf(
```

Comment:
odoo 15/ python 3X  we can write super()._run_wkhtmltopdf()  
"old" nomenclature is necessary if you need a modified "self" e.g. self.with_context()  or self.sudo()
or whatever. but in this case the more concise version is Okay.